### ISSUE TITLE
[Bug] My game crashes when I open any chest

### ISSUE THREAD

**Issue by FutureLight31** [TimVincii/Terrastorage#3]
### What happened?

I am someone who uses a lot of mods, when I add the terrastorage mod to my mods, my game crashes when I open any chest. I play 1.21 fabric. Even though I'm playing single player, I get the error "Connection lost, network protocol error" when I open a chest.
[disconnect-2024-09-25_14.37.00-client.txt](https://github.com/user-attachments/files/17130287/disconnect-2024-09-25_14.37.00-client.txt)

### Terrastorage Version

1.0.1

### Fabric API Version

0.102.0

### Minecraft Version

1.21

### Other Mods

```shell
accurateblockplacement: Accurate Block Placement 1.2.5
		advancementplaques: Advancement Plaques 1.6.6
		adventurez: AdventureZ 1.4.21
		ambientsounds: AmbientSounds 6.0.5
		animal_feeding_trough: Animal Feeding Trough 1.1.1+1.21
		appleskin: AppleSkin 3.0.5+mc1.21
		architectury: Architectury 13.0.6
		armorful: Armorful 3.1.0+1.21
		arrowentitylootdrop: Arrow Entity Loot Drop 1.2.3
		artifacts: Artifacts 12.0.6
			expandability: ExpandAbility 12.0.0
		audioplayer: AudioPlayer 1.21-1.10.2
			admiral: Admiral 0.4.4+1.20.5
			fabric-api-base: Fabric API Base 0.4.42+6573ed8c90
		auto-workstations: Auto Workstations 1.0-rc.25
		axolotlitemfix: Axolotl Item Fix 1.1.9
		badoptimizations: BadOptimizations 2.1.4
		badpackets: Bad Packets 0.8.1
		barteringstation: Bartering Station 21.0.0
		bbm: BetterBoatMovement 2.2.2-1.21+fabric
			dev_nyon_konfig: konfig 2.0.2-1.20.4
		bee_info: MoreBeeInfo 1.3
		better-suggestions: Better Suggestions 1.2.7.1
		better_mcdonalds_mod: Better McDonald's Mod 4.2.0+1.21
		betteranimationscollection: Better Animations Collection 21.0.0
		betterarcheology: Better Archeology 1.2.0-1.21
		bettercombat: Better Combat 2.0.2+1.21.1
		bettercrafting: Craft Better 1.21.1-17-fabric
		betterenchanting: Better Enchanting 0.5.0
		betterf1: BetterF1 1.0
		betterf3: BetterF3 11.0.1
		betterladdersmod: Better Ladders 0.0.3.1.21
		bettermultishot: Better Multishot 2.5.0
			bettermultishot-common: Better Multishot 2.5.0
		betterspawnercontrol: Better Spawner Control 4.6
		betterstats: Better Statistics Screen 3.12.4+fabric-1.21
		biggerenderchests: Bigger Ender Chests 1.0.0
		blur: Blur+ (Fabric) 4.1.0
			midnightlib: MidnightLib 1.5.7
		bombsaway: BombsAway 1.0.0
		boss_music_mod: Boss Music Mod 1.0.0
		camerapture: Camerapture 1.6.2
			io_github_darkxanter_webp-imageio: webp-imageio 0.3.2
		cat_jam: cat_jam 1.2.3
		charmofundying: Charm of Undying 9.0.0+1.21
			spectrelib: SpectreLib 0.17.1+1.21
		chat_heads: Chat Heads 0.12.9
		cherishedworlds: Cherished Worlds 10.0.1+1.21
		cicada: CICADA 0.8.3+1.21-and-above
		cloth-config: Cloth Config v15 15.0.130
			cloth-basic-math: cloth-basic-math 0.6.1
		clumps: Clumps 18.0.0.1
		cobweb: Cobweb 1.1.2
		collective: Collective 7.80
		connectiblechains: Connectible Chains 2.4.2+1.21.1
		coroutil: CoroUtil 1.21.0-1.3.7
		creativecore: CreativeCore 2.12.6
			net_neoforged_bus: bus 7.2.0
		cristellib: Cristel Lib 1.2.8
			blue_endless_jankson: jankson 1.2.3
		ctov: ChoiceTheorem's Overhauled Village 3.5.2
		cutthrough: Cut Through 21.0.0
		danger_close: Danger Close 3.1.1
		dark-loading-screen: Dark Loading Screen 1.6.14
		darkenchanting: Dark Enchanting 1.3.0+1.21
			omegaconfig: Omega Config Architectury 1.5.1
		dogslie: Let Sleeping Dogs Lie 1.3.0
		doors-closed: Doors Closed 1.0.0+1.19.4
		duckling: Duckling 5.0.1
		dungeons_arise: When Dungeons Arise 2.1.60
		dynamiclights: Dynamic Lights 1.8.3+mod
		easyshulkerboxes: Easy Shulker Boxes 21.0.3
			iteminteractions: Item Interactions 21.0.5
		eatinganimationid: Eating Animation 1.21+1.9.72
		effective_invisibility: Effective Invisibility 1.0.1
		eg_particle_interactions: Particle Interactions 0.1.2
		endrem: End Remastered 6.0.0
		entity_model_features: Entity Model Features 2.1.3
		entity_texture_features: Entity Texture Features 6.1.3
			org_apache_httpcomponents_httpmime: httpmime 4.5.10
		entityculling: EntityCulling 1.6.6
		equipmentcompare: Equipment Compare 1.3.12
		expanded_ecosphere: Expanded Ecosphere 3.4.0
		explosiveenhancement: Explosive Enhancement 1.2.3-1.21.0
		fabric-api: Fabric API 0.102.0+1.21
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.68+b5597344d1
			fabric-biome-api-v1: Fabric Biome API (v1) 13.0.29+5bd9f1bcd1
			fabric-block-api-v1: Fabric Block API (v1) 1.0.22+0af3f5a7d1
			fabric-block-view-api-v2: Fabric BlockView API (v2) 1.0.10+6573ed8cd1
			fabric-blockrenderlayer-v1: Fabric BlockRenderLayer Registration (v1) 1.1.52+0af3f5a7d1
			fabric-client-tags-api-v1: Fabric Client Tags 1.1.15+6573ed8cd1
			fabric-command-api-v1: Fabric Command API (v1) 1.2.49+f71b366fd1
			fabric-command-api-v2: Fabric Command API (v2) 2.2.28+6ced4dd9d1
			fabric-commands-v0: Fabric Commands (v0) 0.2.66+df3654b3d1
			fabric-content-registries-v0: Fabric Content Registries (v0) 8.0.16+b5597344d1
			fabric-convention-tags-v1: Fabric Convention Tags 2.0.19+7f945d5bd1
			fabric-convention-tags-v2: Fabric Convention Tags (v2) 2.5.0+c5e2b5c6d1
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.29+0af3f5a7d1
			fabric-data-attachment-api-v1: Fabric Data Attachment API (v1) 1.1.27+6a6dfa19d1
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 20.2.16+16c4ae25d1
			fabric-dimensions-v1: Fabric Dimensions API (v1) 4.0.0+6fc22b99d1
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.6.12+6fc22b99d1
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.7.12+ba9dae06d1
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.53+6ced4dd9d1
			fabric-item-api-v1: Fabric Item API (v1) 11.0.0+afdfc921d1
			fabric-item-group-api-v1: Fabric Item Group API (v1) 4.1.4+78017270d1
			fabric-key-binding-api-v1: Fabric Key Binding API (v1) 1.0.47+0af3f5a7d1
			fabric-keybindings-v0: Fabric Key Bindings (v0) 0.2.45+df3654b3d1
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.3.12+6c1df360d1
			fabric-loot-api-v2: Fabric Loot API (v2) 3.0.14+3f89f5a5d1
			fabric-loot-api-v3: Fabric Loot API (v3) 1.0.2+3f89f5a5d1
			fabric-message-api-v1: Fabric Message API (v1) 6.0.13+6573ed8cd1
			fabric-model-loading-api-v1: Fabric Model Loading API (v1) 2.0.0+fe474d6bd1
			fabric-networking-api-v1: Fabric Networking API (v1) 4.2.2+60c3209bd1
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 15.2.0+a551f7a4d1
			fabric-particles-v1: Fabric Particles (v1) 4.0.2+6573ed8cd1
			fabric-recipe-api-v1: Fabric Recipe API (v1) 5.0.12+65089712d1
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 5.1.2+60c3209bd1
			fabric-renderer-api-v1: Fabric Renderer API (v1) 3.4.0+c705a49cd1
			fabric-renderer-indigo: Fabric Renderer - Indigo 1.7.0+c705a49cd1
			fabric-renderer-registries-v1: Fabric Renderer Registries (v1) 3.2.68+df3654b3d1
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.48+73761d2ed1
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.1.6+b5597344d1
			fabric-rendering-v0: Fabric Rendering (v0) 1.1.71+df3654b3d1
			fabric-rendering-v1: Fabric Rendering (v1) 5.0.5+df16efd0d1
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 4.3.0+8dc279b1d1
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 1.3.0+56599129d1
			fabric-screen-api-v1: Fabric Screen API (v1) 2.0.24+b5597344d1
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.86+b5597344d1
			fabric-sound-api-v1: Fabric Sound API (v1) 1.0.23+6573ed8cd1
			fabric-transfer-api-v1: Fabric Transfer API (v1) 5.1.16+3dccd343d1
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 6.1.0+176f9036d1
		fabric-language-kotlin: Fabric Language Kotlin 1.12.1+kotlin.2.0.20
			org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 2.0.20
			org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 2.0.20
			org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 2.0.20
			org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 2.0.20
			org_jetbrains_kotlinx_atomicfu-jvm: atomicfu-jvm 0.25.0
			org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.8.1
			org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.8.1
			org_jetbrains_kotlinx_kotlinx-datetime-jvm: kotlinx-datetime-jvm 0.6.1
			org_jetbrains_kotlinx_kotlinx-io-bytestring-jvm: kotlinx-io-bytestring-jvm 0.5.3
			org_jetbrains_kotlinx_kotlinx-io-core-jvm: kotlinx-io-core-jvm 0.5.3
			org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.7.1
			org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.7.1
			org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.7.1
		fabricloader: Fabric Loader 0.16.5
			mixinextras: MixinExtras 0.4.1
		fallingleaves: Falling Leaves 1.16.2
		firstperson: FirstPerson 2.4.4
		fishofthieves: Fish of Thieves 4.0.1
		forgeconfigapiport: Forge Config API Port 21.0.8
		formations: Formations 1.0.2
		formationsnether: Formations Nether 1.0.5
		frost: New FrostWalker 1.2.3
		geckolib: GeckoLib 4 4.5.8
		gpu_tape: GPU Tape 1.0.1
		guardvillagers: GuardVillagers 2.1.2-1.21
		held-item-info: Held Item Info 1.7.0
		herdspanic: Herds Panic 1.1.1
		hopobetterunderwaterruins: Hopo Better Underwater Ruins 1.2.0
		iceberg: Iceberg 1.2.4
		ichunutil: iChunUtil 1.0.0
		immediatelyfast: ImmediatelyFast 1.2.21+1.21.1
			net_lenni0451_reflect: Reflect 1.3.4
		indypets: IndyPets - Independent Pets 1.4.1
		iris: Iris 1.8.0-beta.3+mc1.21.1
			io_github_douira_glsl-transformer: glsl-transformer 2.0.1
			org_anarres_jcpp: jcpp 1.4.14
			org_antlr_antlr4-runtime: antlr4-runtime 4.13.1
		jamlib: JamLib 1.0.10+1.21
		java: OpenJDK 64-Bit Server VM 21
		jei: Just Enough Items 19.18.10.218
		justenoughbreeding: Just Enough Breeding 1.4.0
		lios_overhauled_villages: Lio's Overhauled Villages 1.18.2-1.21-v0.0.4
		lithium: Lithium 0.12.7
		lithostitched: Lithostitched 1.3.0
		make_bubbles_pop: Make Bubbles Pop 0.3.0-fabric
		manure: Manure 3.6
		mavapi: More Axolotl Variants API 1.2.1
		mavm: More Axolotl Variants Mod 1.3.0
		medieval_buildings: Medieval Buildings 1.0.2
		minecraft: Minecraft 1.21
		modelfix: Model Gap Fix 1.21-1.6
		modernfix: ModernFix 5.19.0+mc1.21
		modmenu: Mod Menu 11.0.1
		monkeylib538: MonkeyLib538 2.0.1
			monkeylib538-common: MonkeyLib538 2.0.1
		monolib: MonoLib 1.3.0
		monsters_in_the_closet: Monsters in the Closet 1.0.3+1.21
		more_babies: More Babies 2.0.0
		more_bows_and_arrows: More Bows and Arrows 3.0.1
		moremobs: More Mobs 1.5.2+mod
		movement-in-gui: Movement In GUI 1.0.2
		mr_aetherial_islands: Aetherial Islands 1-v1.0.3
		mr_aio_stringmywool: AIO - String my wool 1.2.1
		mr_ashens_betterai: Ashen's Better AI 1.0
		mr_better_lushcaves: Better Lush Caves 1.3.2
		mr_better_mob_drop: Better Mob Drop 1.1.1
		mr_better_trees: Better Trees 1.8.1
		mr_bft: Better Fletching Tables (BFT) 3.0.0
		mr_blood_moon: Horde Nights 1.0.2
		mr_blooming_biosphere: Blooming Biosphere 1.1.2
		mr_cherry_villages: Cherry Grove Villages 1.0.8
		mr_dungeons_andtaverns: Dungeons and Taverns 4.4
		mr_dungeons_andtavernsswamphutoverhaul: Dungeons and Taverns Swamp Hut Overhaul 1-v2
		mr_dynamic_torches: Dynamic Torches 4.5
		mr_horse_trumpet: Horse Trumpet 1.1
		mr_mace: Better Maces 1.0.0
		mr_mine_treasure: Mine Treasure 1.4.6
		mr_nice_villagersremastered: Nice Villagers Remastered 1.0
		mr_scorched: Scorched 1.1.4
		mr_tidal_towns: Tidal Towns 1.3.3
		mr_true_ending: True Ending: Ender Dragon Overhaul 1-v1.1.2d
		mr_upgraded_mobs: Upgraded Mobs 1.4.0
		mr_useful_mobs: Useful Mobs 2.4
		mru: Mineblock's Repeated Utilities 0.4.4+1.21
			yet_another_config_lib_v3: YetAnotherConfigLib 3.5.0+1.21-fabric
				com_twelvemonkeys_common_common-image: common-image 3.10.0
				com_twelvemonkeys_common_common-io: common-io 3.10.0
				com_twelvemonkeys_common_common-lang: common-lang 3.10.0
				com_twelvemonkeys_imageio_imageio-core: imageio-core 3.10.0
				com_twelvemonkeys_imageio_imageio-metadata: imageio-metadata 3.10.0
				com_twelvemonkeys_imageio_imageio-webp: imageio-webp 3.10.0
				org_quiltmc_parsers_gson: gson 0.2.1
				org_quiltmc_parsers_json: json 0.2.1
		narrus_yeetus: Narrus Yeetus 1.1.0
		natures_spirit: Nature's Spirit 1.6.0-1.21
		neutral_animals: Neutral Animals 1.2.6
		no-feeding-baby: No Feeding Baby 1.2.0
		noanimaltemptdelay: No Animal Tempt Delay 1.2
		notenoughanimations: NotEnoughAnimations 1.7.4
		nyfsspiders: Nyf's Spiders 2.3.0
			commonnetworking: Common Network 1.0.11-1.21
		open_sesame: Open Sesame 1.0.0
		pets-not-friendly-fire: Pets Not Friendly Fire 1.0.2
		photography: Photography 1.0.4+mc1.21
		placeholder-api: Placeholder API 2.4.1+1.21
		playeranimator: Player Animator 2.0.0-alpha1+1.21
		plushies: Plushie Mod 1.5.0
		portablehole: Portable Hole 21.0.0
		puzzleslib: Puzzles Lib 21.0.28
		realisticbees: Realistic Bees 4.0
		reeses-sodium-options: Reese's Sodium Options 1.8.0-beta.2+mc1.21.1
		rightclickharvest: Right Click Harvest 4.3.1+1.21
		sculkhornid: SculkHornMod 2.3
		serilumsspawnbundle: Serilum's Spawn Bundle 2.0
			boggedspawn: Bogged Spawn 1.1
			breezespawn: Breeze Spawn 1.1
			cavespiderspawn: Cave Spider Spawn 1.2
			giantspawn: Giant Spawn 5.2
			huskspawn: Husk Spawn 3.6
			mooshroomspawn: Mooshroom Spawn 3.6
			mooshroomtweaks: Mooshroom Tweaks 3.5
			morezombievillagers: More Zombie Villagers 3.6
			skeletonhorsespawn: Skeleton Horse Spawn 4.0
			strayspawn: Stray Spawn 3.7
			zombiehorsespawn: Zombie Horse Spawn 5.1
			zombievillagersfromspawner: Zombie Villagers From Spawner 3.9
		shulkerboxtooltip: Shulker Box Tooltip 5.0.4+1.21
		simplemod: BetterEndCities-Vanilla 1.21
		skinlayers3d: 3d-Skin-Layers 1.6.6
		smallships: Small Ships 2.0.0-b1.5
		smoothscrollingrefurbished: Smooth Scrolling Refurbished 1.1.4
		snuffles: Snuffles 1.2.0+1.21
		sodium: Sodium 0.6.0-beta.1+mc1.21
		sodium-extra: Sodium Extra 0.6.0-beta.1+mc1.21.1
			com_lodborg_interval-tree: interval-tree 1.0.0
		soul_fire_d: Soul Fire'd 5.0.3
		sound_physics_remastered: Sound Physics Remastered 1.21-1.4.5
		soundcontroller: Sound Controller 1.1.3
		stackrefill: Stack Refill 4.5
		stretchy-leash: Stretchy Leash 2.1.0+1.21
		structory: Structory 1.3.5
		superbsteeds: Superb Steeds 1.21-r1
			cardinal-components-base: Cardinal Components API (base) 6.1.1
			cardinal-components-entity: Cardinal Components API (entities) 6.1.1
		supermartijn642configlib: SuperMartijn642's Config Lib 1.1.8
		tameable-foxes: Tameable Foxes 1.0.0
		tcdcommons: TCD Commons API 3.12.2+fabric-1.21
		tectonic: Tectonic 2.4.0
		terrablender: TerraBlender 4.0.0.1
		terrastorage: Terrastorage 1.0.1+1.21-1.21.1
			com_electronwill_night-config_core: core 3.8.1
			com_electronwill_night-config_toml: toml 3.8.1
		the_bumblezone: The Bumblezone - Fabric 7.6.22+1.21-fabric
			athena: Athena 4.0.0
			resourcefullib: Resourceful Lib 3.0.9
				com_teamresourceful_bytecodecs: bytecodecs 1.1.2
				com_teamresourceful_yabn: yabn 1.0.3
		timm: The Immersive Music Mod 1.0.3
		travelersbackpack: Traveler's Backpack fabric-1.21-10.0.4
		traverse: Traverse 8.0.0
			biolith: Biolith 3.0.0-alpha.1
			terraform-biome-remapper-api-v1: Terraform Biome Remapper API (v1) 11.0.0-alpha.1
			terraform-config-api-v1: Terraform Config API (v1) 11.0.0-alpha.1
			terraform-surfaces-api-v1: Terraform Surfaces API (v1) 11.0.0-alpha.1
			terraform-tree-api-v1: Terraform Tree API (v1) 11.0.0-alpha.1
			terraform-wood-api-v1: Terraform Wood API (v1) 11.0.0-alpha.1
			traverse-client: Traverse: Client 8.0.0
			traverse-common: Traverse: Common 8.0.0
			traverse-worldgen: Traverse: World Generation 8.0.0
		trinkets: Trinkets 3.10.0
		variantsandventures: Variants & Ventures 1.0.5
		villagehealthcare: Village Healthcare 1.1.0
		visuality: Visuality 0.7.7+1.21
		voicechat: Simple Voice Chat 1.21-2.5.19
		voidz: VoidZ 1.0.11
		vshorses: Vanilla* Horses 1.0.1+1.21.1
		wakes: Wakes 0.2.5+1.21
			com_github_jdiemke_delaunay-triangulator_delaunaytriangulator: DelaunayTriangulator 1.0.0
			satin: Satin 2.0.0
		waveycapes: WaveyCapes 1.4.8
		weeping_angels: Weeping Angels mc1.21-v1.0.2
		wi_zoom: WI Zoom 1.5-MC1.21
		wsnds: Wither Skeleton No Drop Sword 0.1.1
		wthit: wthit 12.2.3
		xaerominimap: Xaero's Minimap 24.4.0
		xaeroplus: XaeroPlus 2.23.2
			com_github_ben-manes_caffeine_caffeine: caffeine 3.1.8
			net_lenni0451_lambdaevents: LambdaEvents 2.4.2
		xaeroworldmap: Xaero's World Map 1.39.0
		zombieawareness: Zombie Awareness 1.21.0-1.13.2
		zombieproofdoors: Zombie Proof Doors 3.4
```

**Comment by FutureLight31** [comment]
Now my game doesn't crash, thanks for that but there is a small typo in the buttons, I would be very happy if you could fix it. Have a nice day.
![Ekran Alıntısı](https://github.com/user-attachments/assets/2bb042d7-6bef-471b-b7d9-3ad5886f7d93)

**Comment by TimVincii** [comment]
Hi again, thanks for confirming that the game doesn't crash anymore, as for the small typo with the sort items button, it should now be fixed in version 1.0.4, once again, thanks for taking the time to report the bug, and I wish you a great day!
